operators/view_history.py

```diff
@@ -39,6 +39,22 @@ def execute(self, context):
         for prop in selection.__annotations__.keys():
             if hasattr(context.scene.dream_textures_prompt, prop):
                 setattr(context.scene.dream_textures_prompt, prop, getattr(selection, prop))
+            # when the seed of the promt is found in the available image datablocks, use that one in the open image editor
+            # note: when there is more than one image with the seed in it's name, do nothing. Same when no image with that seed is available.
+            if prop == 'seed':
+                seed_string = str(getattr(selection, prop))
+                image_names_no_suffix = [i.name.split('.')[0] for i in bpy.data.images]
+                image_names_no_suffix_seed_count = image_names_no_suffix.count(seed_string)
+                if image_names_no_suffix_seed_count == 1:
```

Comment:
Does an exact match catch it correctly:
```suggestion
                if (existing_image := bpy.data.images.get(seed_string, None)) is not None:
```

Reply:
Unfortunately the proposed change is not equivalent. Consider the following situation. We have made three runs with constant seed of 1096871657 because we wanted to see how the result looked with different samplers. Thus we have three images: 1096871657, 1096871657.001 and 1096871657.002.
Now I want to recall the second promt. With your change, it would select the first of the three images, which is the wrong one. Unfortunately we do not know which of the three is the right image with just the seed. That's why I have decided to count the occurences of a seed in the prefixes of the names of all image datablocks (line 46 and 47) and only select an image if it is the only one with that seed as prefix (line 48). 
When there is more than one occurence, it does not change the current image but reports that more than one has been found (lines 54 and 55).

I am aware that this is a heuristic and not perfect but I think I have an idea for a better solution. The problem is that the unique identifier for datablocks in Blender is the name, which is prone to be changed by the user and sideffects of Blender (.001 .002 .003 etc.). What we can do is the following: After an image is created, run hash() over the datablock and store the result as a custom property of said datablock and store it as well in dream_textures_prompt . That way we can definitely select the correct image no matter how it is named.